File: adt2extn/nul_volume_bars.c
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2025 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

void nul_volume_bars (void)
{
  String_t t; // 14+1 bytes

  t = ExpStrR ((String *)"\x00" "", 14, charmap.bd_heavy_horiz); // "━".."━"

  for (uint8_t i = 0; i < MAX_TRACKS; i++)
    if (channel_flag[chan_pos + i])
      show_str (8 + i * 15, 10 + MAX_PATTERN_ROWS + 1,
                (String *) &t,
                pattern_bckg + pattern_border);
}

File: adt2extn/pas/nul_volume_bars.pas
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2025 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

procedure nul_volume_bars;

var
  chan: Byte;

begin
  For chan := chan_pos to chan_pos+MAX_TRACKS-1 do
    If channel_flag[chan] then
      show_str(08+(chan-PRED(chan_pos)-1)*15,MAX_PATTERN_ROWS+11,
               ExpStrR('',14,#205),
               pattern_bckg+pattern_border);
end;
